From: Ben Hutchings <ben@decadent.org.uk>
Date: Sun, 06 Mar 2016 01:31:13 +0000
Subject: ipv6: Fix ABI change in 4.4.4
Forwarded: not-needed

Change ip6_route_output() back from a static inline wrapper to a proper
extern and exported function.

---
--- a/include/net/ip6_route.h
+++ b/include/net/ip6_route.h
@@ -66,13 +66,8 @@ void ip6_route_input(struct sk_buff *skb
 
 struct dst_entry *ip6_route_output_flags(struct net *net, const struct sock *sk,
 					 struct flowi6 *fl6, int flags);
-
-static inline struct dst_entry *ip6_route_output(struct net *net,
-						 const struct sock *sk,
-						 struct flowi6 *fl6)
-{
-	return ip6_route_output_flags(net, sk, fl6, 0);
-}
+struct dst_entry *ip6_route_output(struct net *net, const struct sock *sk,
+				   struct flowi6 *fl6);
 
 struct dst_entry *ip6_route_lookup(struct net *net, struct flowi6 *fl6,
 				   int flags);
--- a/net/ipv6/route.c
+++ b/net/ipv6/route.c
@@ -1200,6 +1200,13 @@ struct dst_entry *ip6_route_output_flags
 }
 EXPORT_SYMBOL_GPL(ip6_route_output_flags);
 
+struct dst_entry *ip6_route_output(struct net *net, const struct sock *sk,
+				   struct flowi6 *fl6)
+{
+	return ip6_route_output_flags(net, sk, fl6, 0);
+}
+EXPORT_SYMBOL(ip6_route_output);
+
 struct dst_entry *ip6_blackhole_route(struct net *net, struct dst_entry *dst_orig)
 {
 	struct rt6_info *rt, *ort = (struct rt6_info *) dst_orig;
